### ISSUE TITLE
According to the default config settings, how many epochs are typically required to achieve recall@1 above 90

### ISSUE THREAD

**Issue by qwqwqer233** [Skyy93/Sample4Geo#39]
I trained for just 1 epoch with the default configuration, and recall@1 was around 5
Below is the content of my log file.

Model: convnext_base.fb_in22k_ft_in1k_384
{'input_size': (3, 384, 384), 'interpolation': 'bicubic', 'mean': (0.485, 0.456, 0.406), 'std': (0.229, 0.224, 0.225), 'crop_pct': 1.0, 'crop_mode': 'squash'}
GPUs available: 1

Image Size Query: (384, 384)
Image Size Ground: (384, 384)
Mean: (0.485, 0.456, 0.406)
Std:  (0.229, 0.224, 0.225)

Query Images Test: 37854
Gallery Images Test: 951

Scheduler: cosine - max LR: 0.001
Warmup Epochs: 0.1 - Warmup Steps: 946.4000000000001
Train Epochs:  1 - Train Steps:  9464

Shuffle Dataset:
Original Length: 37854 - Length after Shuffle: 37852
Break Counter: 0
Pairs left out of last batch to avoid creating noise: 2
First Element ID: 1094 - Last Element ID: 1342

------------------------------[Epoch: 1]------------------------------
Epoch: 1, Train Loss = 0.913, Lr = 0.000000

------------------------------[Evaluate]------------------------------
Extract Features:
Compute Scores:
Recall@1: 5.9994 - Recall@5: 19.6967 - Recall@10: 29.6904 - Recall@top1: 31.2992 - AP: 9.9101

Shuffle Dataset:
Original Length: 37854 - Length after Shuffle: 37852
Break Counter: 0
Pairs left out of last batch to avoid creating noise: 2
First Element ID: 0945 - Last Element ID: 1501

**Comment by Skyy93** [comment]
Thanks for using our code. According to the training steps in your log I assume that you use a batch size of 4. The InfoNCE loss ist very dependent on high batch sizes to achieve high performance. Which is why we use 128 and in this setting one epoch is enough. 

I am aware that multiple people struggle through to constraint memory. A possible solution could be the implementation of gradient accumulation to calculate the loss on a higher batch sizes.
